File: SpellChecker.java
import java.util.Scanner;
public class SpellChecker{
    public static void printArray(String []corrections){
        for (int j = 0; j < corrections.length; j++){
            System.out.print(corrections[j]+" ");
        }
        System.out.println("");
    }
    public static void main(String []args){
        Words wordDictionary = new Words("english.txt");

        Scanner inputReader = new Scanner(System.in);
        String sentence = inputReader.nextLine();

        String []wordList = sentence.split(" ");
        for (int i = 0; i < wordList.length; i++){
            if (!wordDictionary.isWord(wordList[i])){
                System.out.println("Correcting word " + wordList[i]);
                String []corrections = wordDictionary.getReplaceCorrections(wordList[i]);
                System.out.print("Replace corrections: ");
                printArray(corrections);
                corrections = wordDictionary.getDeleteCorrections(wordList[i]);
                System.out.print("Delete corrections: ");
                printArray(corrections);
                corrections = wordDictionary.getAddCorrections(wordList[i]);
                System.out.print("Add corrections: ");
                printArray(corrections);
            }
        }

    }
}

File: spell_checker.py
from words import Words #Import the py file

english_words = Words('english.txt') #Pass the txt file to the class
sentence = input('Enter a sentence: ') #Enter the sentence to be checked
print(' ')
print(' ')
print(' ')
word_list = sentence.split(' ') #Split the sentence into individual words
for word in word_list: #Loop through every word
    if not english_words.is_word(word.strip()): #Check if the word is present in the file or not
        print('correcting', word) #Begin correcting the word
        print(' ')
        corrections = english_words.get_replace_corrections(word) #First get the word suggetions that can be replaced
        print('Replace suggestions: ', corrections) #Show the replace suggestions to the user
        print(' ')
        corrections = english_words.get_delete_corrections(word) #Then get suggestions that can be shown by deletion
        print('Delete suggestions: ', corrections)
        print(' ')
        corrections = english_words.get_add_corrections(word) #Then get suggestions that can be added by addition of letters
        print('Add suggestions: ', corrections)
        print(' ')
        corrections = english_words.get_add_corrections_without_delete(word)
        print('Without Delete: ', corrections)
        print(' ')
